Packages/vcs/Lib/VTKPlots.py

```diff
@@ -884,8 +885,64 @@ def plot2D(self,data1,data2,tmpl,gm,vtk_backend_grid=None,vtk_backend_geo=None):
           cot.SetValue(i,levs[i])
         cot.SetValue(Nlevs,levs[-1])
         cot.Update()
-        mapper.SetInputConnection(cot.GetOutputPort())
         mappers = []
+        if gm.label=="y":
+            mapper = vtk.vtkLabeledContourMapper()
+        else:
+            mapper = vtk.vtkPolyDataMapper()
+        lut = vtk.vtkLookupTable()
+        lut.SetNumberOfTableValues(1)
+        lut.SetTableValue(0, 0, 0, 0)
+        if gm.label=="y":
+            mapper.GetPolyDataMapper().SetLookupTable(lut)
+            mapper.SetLabelVisibility(1)
+        else:
+            mapper.SetLookupTable(lut)
+
+        # Create text properties.
+        if gm.label=="y":
+         if gm.text or gm.textcolors:
+          colorOverrides = gm.textcolors if gm.textcolors else [None] * len(gm.text)
+          texts = gm.text if gm.text else [None] * len(gm.textcolors)
+          while len(texts)<Nlevs:
+              texts.append(texts[-1])
+          while len(colorOverrides)<Nlevs:
+              colorOverrides.append(colorOverrides[-1])
+          tprops = vtk.vtkTextPropertyCollection()
+          for tc,colorOverride in zip(texts, colorOverrides):
+              if vcs.queries.istextcombined(tc):
```

Comment:
I still think this should be a core vcs function somewhere -- reproducing logic like these every time it's needed is burdensome, error-prone, and difficult to maintain if changes are made to the text specification (every instance of this logic would need to be updated manually).

Not worth delaying this patch, but something to think about.